test/e2e/monitoring.go

```diff
@@ -286,10 +287,18 @@ func validatePodsAndNodes(c clientset.Interface, expectedPods, expectedNodes []s
 
 func testMonitoringUsingHeapsterInfluxdb(c clientset.Interface) {
 	// Check if heapster pods and services are up.
-	expectedPods, err := verifyExpectedRcsExistAndGetExpectedPods(c)
-	framework.ExpectNoError(err)
-	framework.ExpectNoError(expectedServicesExist(c))
-	// TODO: Wait for all pods and services to be running.
+	var expectedPods []string
+	rcErr := fmt.Errorf("Failed to verify expected RCs within timeout")
+	serviceErr := fmt.Errorf("Failed to verify expected services within timeout")
+	for start := time.Now(); time.Since(start) < initializationTimeout; time.Sleep(sleepBetweenAttempts) {
+		expectedPods, rcErr = verifyExpectedRcsExistAndGetExpectedPods(c)
+		serviceErr = expectedServicesExist(c)
+		if rcErr == nil && serviceErr == nil {
+			break
+		}
```

Comment:
I think non-nil errors should be logged as warnings to aid in troubleshooting instead of ignoring them completely.